### PR TITLE
Stats: Cleanup unused GMB and Upwork stats banners nudges

### PR DESCRIPTION
**Pull request by tyxla**
This is a follow-up to #40301 that removes some more of the unused code.

#### Changes proposed in this Pull Request

* Remove unused Google My Business stats nudge
* Remove unused Upwork stats nudge
* Remove unused components and state, related with the above nudges.

#### Testing instructions

* Verify none of the removed code is used anywhere.

### CODE REVIEW

File: client/state/preferences/schema.js [hunk 1]
```diff
@@ -13,20 +13,6 @@ export const remoteValuesSchema = {
...
-				items: {
-					type: 'object',
-					properties: {
-						dismissedAt: { type: 'number', minimum: 0 },
-						type: { type: 'string', enum: [ 'dismiss' ] },
-					},
-					required: [ 'dismissedAt', 'type' ],
-				},
-			},
-		},
```
Comment (jsnajdr) [inline client/state/preferences/schema.js:29]: This schema file also defines the `upwork-dismissible-banner` pref, which should be also removed in this PR.

**Review by jsnajdr — APPROVED** [review]
Looks good, except there is one more thing to be removed from the preferences schema file.

File: client/state/preferences/schema.js (hunk 1 above)
Comment (tyxla) [reply client/state/preferences/schema.js:29]: I thought the same and was going to remove it when working on this PR, but then I noticed that it's still used by the `isUpworkBannerDismissed` selector and the `dismissBanner` action creator, and they're still used by the `blocks/upwork-banner` component.

**Comment by tyxla** [comment]
Thank you @jsnajdr 🙌

**Pull request merged**